File: peel-core/src/main/java/eu/stratosphere/peel/core/cli/command/Command.java
package eu.stratosphere.peel.core.cli.command;

import net.sourceforge.argparse4j.inf.Namespace;
import net.sourceforge.argparse4j.inf.Subparser;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.context.ApplicationContext;

abstract public class Command {

    protected final Logger logger = LoggerFactory.getLogger(this.getClass());

    /**
     * Returns a constant name for this command.
     * 
     * @return The unique name for this command
     */
    abstract public String name();

    /**
     * Returns a constant one-sentence help for this command.
     * 
     * @return The help message for this command
     */
    abstract public String help();

    /**
     * Registers the arguments and options to the CLI sub-parser for this command.
     */
    abstract public void register(Subparser parser);

    /**
     * Configures the selected command using the parsed CLI arguments.
     * 
     * @param ns The parsed CLI arguments namespace.
     */
    abstract public void configure(Namespace ns);

    /**
     * Runs the command.
     * 
     * @param context The application context
     */
    abstract public void run(ApplicationContext context);
}

File: peel-core/src/main/scala/eu/stratosphere/peel/core/cli/command/Command.scala
package eu.stratosphere.peel.core.cli.command

import net.sourceforge.argparse4j.inf.{Namespace, Subparser}
import org.slf4j.LoggerFactory
import org.springframework.context.ApplicationContext

abstract class Command {

  val logger = LoggerFactory.getLogger(this.getClass)

  /** The unique name for this command. */
  val name: String

  /** A one-sentence help for this command. */
  val help: String

  /** Registers the arguments and options to the CLI sub-parser for this command.
    *
    * @param parser The parent parser.
    */
  def register(parser: Subparser): Unit

  /** Configures the selected command using the parsed CLI arguments.
    *
    * @param ns The parsed CLI arguments namespace.
    */
  def configure(ns: Namespace): Unit

  /** Runs the command.
    *
    * @param context The application context.
    */
  def run(context: ApplicationContext)
}
